### DISCUSSION TITLE
Total ascend for a given time-period / drive

### DISCUSSION THREAD

**Discussion by zwuba** [teslamate-org/teslamate#2558] [Ideas]
I'm coming from a cycling background and as I live up on a mountain "total ascend" is my absolute most important metric.
Till now I couldn't find such a number anywhere in Teslamate but it would be so much important for me as just seeing my consumption being up in yellow-orange seems to be weird for some other's when I show them ...
but when I tell them I have about 250m of ascend each and every day when I go to work it's a totally different story.

Well all that said I think most of the "mountain guys" will know what I'm talking about and if that would be something that could be integrated it would be awesome ;)

**Comment by init6-git** [comment]
The elevation data is captured, so you should be able create your own dashboard. The tricky bit will be working out what you mean by total ascend. What formula are you currently using e.g. for your cycling?

**Reply by zwuba** [reply]
It's a good question what's the formula ... I don't exactly know as this is embedded in almost every cycling device from Garmin, Polar or whatever other brand.
IMHO it should be pretty simple by having a fixed sampling-period (eg 10sec for cycling) which checks for a possible positive elevation change and this delta is being summed up over time.
Of course this simple equation will become more and more incorrect the hight velocity and elevation-gain gets and there might be more sophisticated calculations :)

I honestly have no idea how I could get this kind of data out of the data easily ... maybe someone of the PROs can help here :)

**Reply by init6-git** [reply]
Easy enough to add up the positive changes between datapoints*, but what happens when you go downhill and then back up? If you are only concerned with positive changes then here's the sql for working out the gain in a particular drive. Bung it on a Stat panel in the drive details dashboard:
```
with data AS (
SELECT
	$__time(date),
	elevation, 
  LAG(elevation,1) OVER (order by date) previous_elevation

FROM
	positions
WHERE
 car_id = $car_id AND
 $__timeFilter(date)
ORDER BY
	date ASC
	)
	
select 
        convert_m(sum (elevation-previous_elevation), '$alternative_length_unit') as "Feet gained"
        
       from data
```
       

Probably neater ways to do it.

* The data is streamed in intervals of less than 1 second. Bear in mind that you can cover 1/4 mile in 11 seconds, then a 10 second interval is probably too chunky. The car's a wee bit faster than a bike :)

**Comment by zwuba** [comment]
Wow that's just awesome - thank you so much!!
I hope to get enough time this weekend to learn where and how to embed this
SQL Statement :)

... oh and yes you exactly nailed the point in your last sentence -
absolutely agree - but I do still think this number comes on very handy
when comparing trips especially especially when living in the alps where a
50k Autobahn ride can easily add up 500meters of elevation where the same
ride in the states might not even come close to 50 :)

Last question - is there a manual/document in place where I can learn by
myself where and how to integrate your shiny new SQL statement?

**Reply by init6-git** [reply]
Google Grafana.
You'll want to create a new dashboard and add that in as a panel. Easiest way is to duplicate the Drive Details dashboard and add the panel to that. You can't amend the existing Teslamate dashboards as they can sometimes get overwritten with upgrades. IIRC the permissions might not even let you do this.
Do you ever go to Teslamotorsclub.com forums? In the UK&I forum there is a Teslamate thread and I've told people a few times how to create these dashboards. If you're on there bung a post on the end of the thread and I'll ping you some instructions.

To save you searching: https://teslamotorsclub.com/tmc/threads/teslamate-megathread.191537/page-105
